tests/unit/bfs_common.hpp

```diff
@@ -440,3 +450,213 @@ void BfsTest(Database *db, int lower_bound, int upper_bound, memgraph::query::Ed
+        case memgraph::query::EdgeAtom::Direction::BOTH:
+          EXPECT_EQ(results.size(), 30);
+          break;
+      }
+      break;
+    case FineGrainedTestType::ALL_DENIED:
+      switch (direction) {
+        case memgraph::query::EdgeAtom::Direction::IN:
+          EXPECT_EQ(results.size(), 0);
+          break;
+        case memgraph::query::EdgeAtom::Direction::OUT:
+          EXPECT_EQ(results.size(), 0);
+          break;
+        case memgraph::query::EdgeAtom::Direction::BOTH:
+          EXPECT_EQ(results.size(), 0);
+          break;
+      }
+      break;
+    case FineGrainedTestType::EDGE_TYPE_A_DENIED:
+      switch (direction) {
+        case memgraph::query::EdgeAtom::Direction::IN:
+          EXPECT_EQ(results.size(), 4);
+          break;
+        case memgraph::query::EdgeAtom::Direction::OUT:
+          EXPECT_EQ(results.size(), 4);
+          break;
+        case memgraph::query::EdgeAtom::Direction::BOTH:
+          EXPECT_EQ(results.size(), 8);
+          break;
+      }
+      break;
+    case FineGrainedTestType::EDGE_TYPE_B_DENIED:
+      switch (direction) {
+        case memgraph::query::EdgeAtom::Direction::IN:
+          EXPECT_EQ(results.size(), 8);
+          break;
+        case memgraph::query::EdgeAtom::Direction::OUT:
+          EXPECT_EQ(results.size(), 8);
+          break;
+        case memgraph::query::EdgeAtom::Direction::BOTH:
+          EXPECT_EQ(results.size(), 20);
+          break;
+      }
+      break;
+    case FineGrainedTestType::LABEL_0_DENIED:
+      switch (direction) {
+        case memgraph::query::EdgeAtom::Direction::IN:
+          if (known_sink) {
+            EXPECT_EQ(results.size(), 6);
+          } else {
+            EXPECT_EQ(results.size(), 9);
+          }
+          break;
+        case memgraph::query::EdgeAtom::Direction::OUT:
+          if (known_sink) {
+            EXPECT_EQ(results.size(), 6);
+          } else {
+            EXPECT_EQ(results.size(), 13);
+          }
+          break;
+        case memgraph::query::EdgeAtom::Direction::BOTH:
+          if (known_sink) {
+            EXPECT_EQ(results.size(), 6);
+          } else {
+            EXPECT_EQ(results.size(), 13);
+          }
+          break;
+      }
+      break;
+    case FineGrainedTestType::LABEL_3_DENIED:
+      switch (direction) {
+        case memgraph::query::EdgeAtom::Direction::IN:
+          if (known_sink) {
+            EXPECT_EQ(results.size(), 9);
+          } else {
+            EXPECT_EQ(results.size(), 13);
+          }
+          break;
+        case memgraph::query::EdgeAtom::Direction::OUT:
+          if (known_sink) {
+            EXPECT_EQ(results.size(), 9);
+          } else {
+            EXPECT_EQ(results.size(), 12);
+          }
+          break;
+        case memgraph::query::EdgeAtom::Direction::BOTH:
+          if (known_sink) {
+            EXPECT_EQ(results.size(), 12);
+          } else {
+            EXPECT_EQ(results.size(), 16);
+          }
+          break;
```

Comment:
I think you should as well here make lists and check for individual inputs, and not just sizes

Reply:
https://github.com/memgraph/memgraph/blob/T0997-MG-authorization-on-paths/tests/unit/bfs_common.hpp also i changed this and git didn't catch differences